Source/DafnyLanguageServer.Test/Various/ConcurrentInteractionsTest.cs

```diff
@@ -73,6 +84,86 @@ public class ConcurrentInteractionsTest : DafnyLanguageServerTestBase {
       Assert.AreEqual(1, diagnostics.Length);
     }
 
+    [TestMethod, Timeout(MaxTestExecutionTimeMs)]
+    public async Task VerificationErrorDetectedAfterCanceledSave() {
+      // Create a document that'll be slightly slow to verify
+      var source = @"
+method Multiply(x: bv10, y: bv10) returns (product: bv10)
+  requires y >= 0 && x >= 0
+  decreases y
+  ensures product == x * y && product >= 0
+{
+  if y == 0 {
+    product := 0;
+  } else {
+    var step := Multiply(x, y - 1);
+    product := x + step;
+  }
+}".TrimStart();
+      var failSource = @"method Contradiction() { assert false; }";
+      await SetUp(new Dictionary<string, string>() {
+        { $"{DocumentOptions.Section}:{nameof(DocumentOptions.Verify)}", nameof(AutoVerification.OnSave) }
+      });
+      var documentItem = CreateTestDocument(source);
+      await client.OpenDocumentAndWaitAsync(documentItem, CancellationTokenWithHighTimeout);
+
+      DidChangeTextDocumentParams MakeChange(int? version, Range range, string text) {
+        return new DidChangeTextDocumentParams {
```

Comment:
Why not make the call `client.DidChangeTextDocument` here already?

Reply:
It would slightly shorten the code, but I tend to like to keep pure computation or data construction separate from effect-producing operations. I don't have a super strong opinion about that, though.